torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2692,13 +2689,17 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
   if (avoidRecordStreams) {
     work->stashed_for_allocator_safety_ =
         std::make_shared<std::vector<at::Tensor>>();
-    work->stashed_for_allocator_safety_->push_back(input);
+    for (const auto& input : inputs) {
+      work->stashed_for_allocator_safety_->push_back(input);
+    }
   }
 
   at::cuda::OptionalCUDAGuard gpuGuard(device);
 
   if (nanCheck) {
-    checkForNan(input, ncclStream);
+    for (const auto& input : inputs) {
+      checkForNan(input, ncclStream);
```

Comment:
hold on a sec- are we re-introducing a NaN-check bug here, where some 'empty' inputs could be Nan-checked?

Reply:
no, for this case if it is for the inputs of all-to-all. What is our fix to the empty inputs? To just use inputs[0]?

Reply:
@wconstab what do you mean by empty inputs? like dummy inputs in non-root ranks?